### PR TITLE
fix: CheckboxName and SwitchName rule

### PR DESCRIPTION
**Pull request by devchan4188**
Closes: #97
## Requester Review Items

- [x] Ensure the Pull Request is into develop.
- [x] Test coverage has not gone down.
- [x] Documentation updated or not needed. (Wiki, Issues, etc)
- [x] Angular Type leads to proper Semantic Version.

## Dev Team Review Items

To be filled out by @dequelabs/mobile.

- [x] Ensure the Requester did not lie. Chastise them politely if they did.
- [x] Code Quality review.
- [x] Changes to Axe*.java classes are minimal, appropriate, and versioned properly.
  - [x] Serialized Axe objects have not changed.
  - [x] Public API has not changed.

## Code Owner Review Items

To be filled out by @dequelabs/mobileadmin.

- [x] Any Rules package changes lead to proper Semantic Version.
- [x] Security Review.

## Merger Review Items

- [x] Ensure commit message matches title before squashing.

### CODE REVIEW

File: src/main/java/com/deque/axe/android/rules/hierarchy/SwitchName.java [hunk 1]
```diff
@@ -1,11 +1,33 @@
...
+
+    super.collectProps(axeView, axeProps);
+  }
+
+  @Override
+  public String runRule(AxeProps axeProps) {
+    final String visibleText = axeProps.get(AxeProps.Name.VISIBLE_TEXT, String.class);
+
+    if (!AxeTextUtils.isNullOrEmpty(visibleText)
+            && !visibleText.equalsIgnoreCase("on") && !visibleText.equalsIgnoreCase("off")) {
```
Comment (cyono) [inline src/main/java/com/deque/axe/android/rules/hierarchy/SwitchName.java]: Extract this into a method with a semantic name that clarifies, in perpetuity, _why_ this is valid.

File: src/main/java/com/deque/axe/android/rules/hierarchy/CheckBoxName.java [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
+    axeProps.put(AxeProps.Name.VISIBLE_TEXT, axeView.text);
+
+    super.collectProps(axeView, axeProps);
+  }
+
+  @Override
+  public String runRule(AxeProps axeProps) {
+    final String visibleText = axeProps.get(AxeProps.Name.VISIBLE_TEXT, String.class);
+
+    if (!AxeTextUtils.isNullOrEmpty(visibleText)) {
```
Comment (cyono) [inline src/main/java/com/deque/axe/android/rules/hierarchy/CheckBoxName.java]: This short-circuits the label checking logic.

File: src/main/java/com/deque/axe/android/rules/hierarchy/SwitchName.java (hunk 1 above)
Comment (chriscm2006) [reply src/main/java/com/deque/axe/android/rules/hierarchy/SwitchName.java]: I think this belongs in AxeView. I would propose adding a method that says `containsNonStateText()` or `hasNameText()` I'm having trouble naming this method.

Comment (cyono) [reply src/main/java/com/deque/axe/android/rules/hierarchy/SwitchName.java]: Agreed. How about `hasImmutableText()` or `hasOperatingSystemText()` ?

**Comment by devchan4188** [comment]
Closing this PR for now since it is blocked on `jsonTestSpec` fix. Will reopen this once `jsonTestSpec` is fixed.

**Pull request merged**